suite-common/wallet-core/src/transactions/transactionsReducer.ts

```diff
@@ -200,16 +202,37 @@ export const selectTransactionFirstTargetAddress = memoizeWithArgs(
     },
 );
 
-export const selectTransactionFirstInputAddress = memoizeWithArgs(
+const selectTransactionTargetAddresses = memoizeWithArgs(
     (state: TransactionsRootState, txid: string, accountKey: AccountKey) => {
+        const transactionTargets = selectTransactionTargets(state, txid, accountKey);
+        if (G.isNullable(transactionTargets)) return [];
+
+        return mapTransactionInputsOutputsToAddresses(transactionTargets);
+    },
+);
+
+export const selectTransactionInputAddresses = memoizeWithArgs(
+    (state: TransactionsRootState, txid: string, accountKey: AccountKey): string[] => {
         const transaction = selectTransactionByTxidAndAccountKey(state, txid, accountKey);
-        return transaction?.details?.vin?.[0].addresses?.[0];
+
+        if (G.isNullable(transaction)) return [];
+
+        const inputAddresses = mapTransactionInputsOutputsToAddresses(transaction.details.vin);
+        const targetAddresses = selectTransactionTargetAddresses(state, txid, accountKey);
+
+        return sortTargetAddressesToBeginning(inputAddresses, targetAddresses);
     },
 );
 
-export const selectTransactionFirstOutputAddress = memoizeWithArgs(
-    (state: TransactionsRootState, txid: string, accountKey: AccountKey) => {
+export const selectTransactionOutputAddresses = memoizeWithArgs(
+    (state: TransactionsRootState, txid: string, accountKey: AccountKey): string[] => {
         const transaction = selectTransactionByTxidAndAccountKey(state, txid, accountKey);
-        return transaction?.details?.vout?.[0].addresses?.[0];
+
+        if (G.isNullable(transaction)) return [];
+
+        const outputAddresses = mapTransactionInputsOutputsToAddresses(transaction.details.vout);
+        const targetAddresses = selectTransactionTargetAddresses(state, txid, accountKey);
+
+        return sortTargetAddressesToBeginning(outputAddresses, targetAddresses);
```

Comment:
same here. I see you used it in utils functions. But it's just a nit. Feel free to resolve as it doesn't bring any benefit, it's just a matter of style